core/src/worker/workflow/wft_poller.rs

```diff
@@ -0,0 +1,86 @@
+use crate::{pollers::BoxedWFPoller, protosext::ValidPollWFTQResponse, MetricsContext};
+use futures::{stream, Stream};
+use temporal_sdk_core_protos::temporal::api::workflowservice::v1::PollWorkflowTaskQueueResponse;
+
+pub(crate) fn new_wft_poller(
+    poller: BoxedWFPoller,
+    metrics: MetricsContext,
+) -> impl Stream<Item = ValidPollWFTQResponse> {
+    stream::unfold((poller, metrics), |(poller, metrics)| async move {
+        loop {
+            match poller.poll().await {
+                Some(Ok(wft)) => {
+                    if wft == PollWorkflowTaskQueueResponse::default() {
+                        // We get the default proto in the event that the long poll times out.
+                        debug!("Poll wft timeout");
+                        metrics.wf_tq_poll_empty();
+                        continue;
+                    }
+                    if let Some(dur) = wft.sched_to_start() {
+                        metrics.wf_task_sched_to_start_latency(dur);
+                    }
+                    let work = match validate_wft(wft) {
+                        Ok(w) => w,
+                        Err(e) => {
+                            error!(error=?e, "Server returned an unparseable workflow task");
+                            continue;
+                        }
+                    };
+                    metrics.wf_tq_poll_ok();
+                    return Some((work, (poller, metrics)));
+                }
+                Some(Err(e)) => {
+                    warn!(error=?e, "Error while polling for workflow tasks");
```

Comment:
Metrics?

Reply:
There is no specific metric for task queue poll failure. There is one for all grpc errors, which is handled inside the client.